thunder-service/database/oci_recommendations.go

```diff
@@ -0,0 +1,163 @@
+			recToDB,
+		)
+	if err != nil {
+		return utils.NewError(err, "DB ERROR")
+	}
+
+	return nil
+}
+
+func (md *MongoDatabase) GetOciRecommendationsByProfiles(profileIDs []string) ([]model.OciRecommendation, error) {
+	ctx := context.TODO()
+
+	inCondition := bson.M{"$in": profileIDs}
+	filter := bson.M{"profileID": inCondition}
+
+	cur, err := md.Client.Database(md.Config.Mongodb.DBName).Collection(OciRecommendation_collection).Find(ctx, filter)
+	if err != nil {
+		return nil, utils.NewError(cur.Err(), "DB ERROR")
+	}
+
+	ercoleRecommendations := make([]model.OciRecommendation, 0)
+	if err := cur.All(context.TODO(), &ercoleRecommendations); err != nil {
+		return nil, utils.NewError(err, "DB ERROR")
+	}
+
+	if err := cur.Err(); err != nil {
+		return nil, utils.NewError(err, "DB ERROR")
+	}
+
+	return ercoleRecommendations, nil
+}
+
+func (md *MongoDatabase) GetOciRecommendations(profileIDs []string) ([]model.OciRecommendation, error) {
+	ctx := context.TODO()
+
+	findOptions := options.Find()
+	findOptions.SetSort(bson.M{"seqValue": -1})
+
+	cur, err := md.Client.Database(md.Config.Mongodb.DBName).Collection(OciRecommendation_collection).Find(ctx, bson.D{}, findOptions)
+	if err != nil {
+		return nil, utils.NewError(err, "DB ERROR")
+	}
+
+	ociRecommendations := make([]model.OciRecommendation, 0)
+	if err := cur.All(context.TODO(), &ociRecommendations); err != nil {
+		return nil, utils.NewError(err, "DB ERROR")
+	}
+
+	if err := cur.Err(); err != nil {
+		return nil, utils.NewError(err, "DB ERROR")
+	}
+
+	inCondition := bson.M{"$in": profileIDs}
+
+	filter := bson.M{"seqValue": ociRecommendations[0].SeqValue, "profileID": inCondition}
+
+	cur1, err1 := md.Client.Database(md.Config.Mongodb.DBName).Collection(OciRecommendation_collection).Find(ctx, filter)
+	if err1 != nil {
+		return nil, utils.NewError(err, "DB ERROR")
+	}
+
+	ociRecommendation1 := make([]model.OciRecommendation, 0)
+	if err := cur1.All(context.TODO(), &ociRecommendation1); err != nil {
+		return nil, utils.NewError(err, "DB ERROR")
+	}
+
+	return ociRecommendation1, nil
+}
+
+func (md *MongoDatabase) DeleteOldOciRecommendations(dateFrom time.Time) error {
+	_, err := md.Client.Database(md.Config.Mongodb.DBName).Collection(OciRecommendation_collection).
+		DeleteMany(
+			context.TODO(),
+			bson.M{"createdAt": bson.M{"$lt": dateFrom}},
+		)
+	if err != nil {
+		return utils.NewError(err, "DB ERROR")
+	}
+
+	return nil
+}
+
+func (md *MongoDatabase) GetLastOciSeqValue() (uint64, error) {
+	ctx := context.TODO()
+
+	findOptions := options.Find()
+	findOptions.SetSort(bson.M{"seqValue": -1})
+
+	cur, err := md.Client.Database(md.Config.Mongodb.DBName).Collection(OciRecommendation_collection).Find(ctx, bson.D{}, findOptions)
+	if err != nil {
+		return 999999, utils.NewError(err, "DB ERROR")
+	}
+
+	ociRecommendations := make([]model.OciRecommendation, 0)
+	if err := cur.All(context.TODO(), &ociRecommendations); err != nil {
+		return 999999, utils.NewError(err, "DB ERROR")
```

Comment:
Could you try to change 999999 with math.MaxUint64?

Reply:
Fixed